subprojects/core/src/main/java/org/gradle/api/internal/component/BuildableJavaComponent.java

```diff
@@ -31,4 +32,6 @@
     FileCollection getRuntimeClasspath();
 
     Configuration getCompileDependencies();
+
+    ProjectInternal getProject();
```

Comment:
I think we should avoid adding this, just because it brings in the kitchen sink.

Reply:
Fair enough.  Removed.